Building with `-DENABLE_SHARED_LIB=OFF` fails.

```
[ 97%] Linking C static library libnghttp3.a
[ 97%] Built target nghttp3_static
1 warning generated.
2 warnings generated.
[100%] Linking CXX executable qpack
ld: library not found for -lnghttp3
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [examples/qpack] Error 1
make[1]: *** [examples/CMakeFiles/qpack.dir/all] Error 2
make: *** [all] Error 2
```

Maybe I should disable building `qpack` binary?

--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -28,9 +28,15 @@ if(ENABLE_EXAMPLES)
     ${CMAKE_BINARY_DIR}/lib/includes
   )
 
-  link_libraries(
-    nghttp3
-  )
+  if(ENABLE_SHARED_LIB)
+    link_libraries(
+      nghttp3
+    )
+  elseif(HAVE_CUNIT OR ENABLE_STATIC_LIB)
+    link_libraries(
+      nghttp3_static
+    )
+  endif()
 
   set(qpack_SOURCES
     qpack.cc

